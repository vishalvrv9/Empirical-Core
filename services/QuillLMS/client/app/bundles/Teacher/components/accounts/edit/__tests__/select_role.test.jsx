import React from 'react';
import { shallow } from 'enzyme';

import SelectRole from '../select_role';

describe('SelectRole component', () => {

  it('should render', () => {
    const wrapper = shallow(
      <SelectRole role={'teacher'} updateRole={() => null} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should trigger the updateRole prop on select value change', () => {
    const mockSelect = jest.fn();
    const wrapper = shallow(
      <SelectRole role={'teacher'} updateRole={mockSelect} />
    );
    wrapper.find('select').simulate('change', { target: { value : 'staff' }});
    expect(mockSelect.mock.calls).toHaveLength(1);
  });

});
