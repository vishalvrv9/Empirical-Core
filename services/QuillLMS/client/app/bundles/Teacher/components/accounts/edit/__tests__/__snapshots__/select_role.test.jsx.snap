// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectRole component should render 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-2 form-label"
  >
    Role
  </div>
  <div
    className="col-xs-4"
  >
    <select
      onChange={[Function]}
      value="teacher"
    >
      <option
        value="teacher"
      >
        teacher
      </option>
      <option
        value="student"
      >
        student
      </option>
      <option
        value="admin"
      >
        admin
      </option>
      <option
        value="staff"
      >
        staff
      </option>
    </select>
  </div>
  <div
    className="col-xs-4 errors"
  />
</div>
`;
